docker/dev/gitlab/docker-compose.yml

```diff
@@ -0,0 +1,37 @@
+version: "3.9"
+
+services:
+  gitlab:
+    image: gitlab/gitlab-ce:latest
+    restart: no
+    volumes:
+      - gitlab-etc:/etc/gitlab
+      - gitlab-logs:/var/log/gitlab
+      - gitlab-opt:/var/opt/gitlab
+      # Linux
+      - /etc/ssl/certs:/etc/ssl/certs:ro
+      - /usr/local/share/ca-certificates:/usr/local/share/ca-certificates:ro
+      # In case of MacOS, you need to mount the certs from the host machine
+      # having them previously bundled via `sudo update-ca-certificates`
+      #
+      # - ~/.step/certs:/etc/ssl/certs
+      # - ~/.step/certs:/usr/local/share/ca-certificates
+    networks:
+      - external
+    extra_hosts:
+      - "openproject.local:host-gateway"
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.gitlab.rule=Host(`gitlab.local`)"
+      - "traefik.http.routers.gitlab.entrypoints=websecure"
+      - "traefik.http.services.gitlab.loadbalancer.server.port=80"
```

Comment:
Stupid question: why we need a new service and loadBalancer defined?

Reply:
@apfohl has a better handle on this but my (very basic) understanding is a dedicated load balancer service for `gitlab` is able to designate traffic to one or more gitlab instances- in dev we'll mainly just use one instance. I reckon it's also better to have split traffic between other sservices

Reply:
It's way easier then you might think. In traefik routing, the loadbalancer stage is always involved. And it is also the point where you can configure to which port the routing should go. Because the container is exposing three ports it will use the lowest by default which is 22. Therefore it's necessary to tell the loadbalancer stage to use port 80. ;)

Reply:
And it's not a separate service.